plugins/spi/SPIOutput.cpp

```diff
@@ -180,26 +185,67 @@ SPIOutput::SPIOutput(const UID &uid, SPIBackendInterface *backend,
   m_spi_device_name = FilenameFromPathOrPath(m_backend->DevicePath());
 
   PersonalityCollection::PersonalityList personalities;
-  personalities.push_back(Personality(m_pixel_count * WS2801_SLOTS_PER_PIXEL,
-                                      "WS2801 Individual Control"));
-  personalities.push_back(Personality(WS2801_SLOTS_PER_PIXEL,
-                                      "WS2801 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * LPD8806_SLOTS_PER_PIXEL,
-                                      "LPD8806 Individual Control"));
-  personalities.push_back(Personality(LPD8806_SLOTS_PER_PIXEL,
-                                      "LPD8806 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * P9813_SLOTS_PER_PIXEL,
-                                      "P9813 Individual Control"));
-  personalities.push_back(Personality(P9813_SLOTS_PER_PIXEL,
-                                      "P9813 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * APA102_SLOTS_PER_PIXEL,
-                                      "APA102 Individual Control"));
-  personalities.push_back(Personality(APA102_SLOTS_PER_PIXEL,
-                                      "APA102 Combined Control"));
+  // personality description is max 32 characters
+
+  personalities.insert(personalities.begin() + PERS_WS2801_INDIVIDUAL - 1,
+    Personality(m_pixel_count * WS2801_SLOTS_PER_PIXEL,
+      "WS2801 Individual Control"));
+  personalities.insert(personalities.begin() + PERS_WS2801_COMBINED - 1,
+    Personality(WS2801_SLOTS_PER_PIXEL,
+      "WS2801 Combined Control"));
+
+  personalities.insert(personalities.begin() + PERS_LDP8806_INDIVIDUAL - 1,
+    Personality(m_pixel_count * LPD8806_SLOTS_PER_PIXEL,
+      "LPD8806 Individual Control"));
+  personalities.insert(personalities.begin() + PERS_LDP8806_COMBINED - 1,
+    Personality(LPD8806_SLOTS_PER_PIXEL,
+      "LPD8806 Combined Control"));
+
+  personalities.insert(personalities.begin() + PERS_P9813_INDIVIDUAL - 1,
+    Personality(m_pixel_count * P9813_SLOTS_PER_PIXEL,
+      "P9813 Individual Control"));
+  personalities.insert(personalities.begin() + PERS_P9813_COMBINED - 1,
+    Personality(P9813_SLOTS_PER_PIXEL,
+      "P9813 Combined Control"));
+
+  personalities.insert(personalities.begin() + PERS_APA102_INDIVIDUAL - 1,
+    Personality(m_pixel_count * APA102_SLOTS_PER_PIXEL,
+      "APA102 Individual Control"));
+
+  ola::rdm::SlotDataCollection::SlotDataList sd_APA102_COMBINED;
+  sd_APA102_COMBINED.push_back(
+    ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_RED, 0));
```

Comment:
Indent to four please.

Reply:
you mean so:
```
  sd_APA102_COMBINED.push_back(
      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_RED, 0));
```
i thought that the indentation rule is 2 spaces.
or is it 2spaces - and on continuing lines 4spaces?
sorry to mess up this things!

Reply:
Yeah, we indent by two (apart from public:/private:, which are one where we've bothered/in some cases), but four for a continuing line.

Reply:
Done.